stetho/src/main/java/com/facebook/stetho/inspector/network/ResponseBodyFileManager.java

```diff
@@ -69,6 +89,25 @@ private static String readContentsAsUTF8(InputStream in) throws IOException {
     return out.toString("UTF-8");
   }
 
+  private String prettyPrintContentWithTimeOut(
+      AsyncPrettyPrinter asyncPrettyPrinter,
+      InputStream in) throws IOException {
+    AsyncPrettyPrintingTask prettyPrintingTask = new AsyncPrettyPrintingTask(
+        in,
+        asyncPrettyPrinter);
+    ExecutorService executor = Executors.newSingleThreadExecutor();
+    Future<String> future = executor.submit(prettyPrintingTask);
+    String prettifiedContent = "";
+    try {
+      prettifiedContent = future.get(10, TimeUnit.SECONDS);
+    } catch (TimeoutException | InterruptedException | ExecutionException e) {
+      future.cancel(true);
+      prettifiedContent = "Failed to pretty print data\n" + readContentsAsUTF8(in);
+    }
+    executor.shutdownNow();
```

Comment:
You don't need this, but you can theoretically shut it down if you hook into `Network`'s cleanup code on the last peer disconnecting.